packages/ember-glimmer/lib/utils/references.js

```diff
@@ -53,6 +53,25 @@ export class UpdatableReference extends RootReference {
   }
 }
 
+export class GetHelperReference {
+  constructor(sourceReference, pathReference) {
+    this.sourceReference = sourceReference;
+    this.pathReference = pathReference;
+  }
+
+  isDirty() { return true; }
+
+  value() {
+    return this.sourceReference.get(this.pathReference.value()).value();
```

Comment:
Since we are returning a value immediately, there is no need to go through `get` on the source reference. Something like your original implementation...

```
let object = this.sourceReference.value();
let path = this.pathReference.value();

return get(object, path);
```

...should be enough (might need some null guards here and there). In general, it is only worth it to produce a reference (`.get` returns a new `PathReference`) if you are going to hold onto it for and use it again in the future. The point of a reference is that you can save it off and query it over and over again to get the latest value.